Simply testing for Conda MOOSE, update libmesh

milljm opened the pull request:
Remove most testing being done during Conda MOOSE package creation. Testing will be implemented as a separate target. This will allow for quicker package creation when things go awry.

Closes #25369

---

From @roystgnr:
libMesh submodule update added, to minimize the number of mamba updates users will get hit with:

- NetCDF upgrade to 4.9.2, now incorporating NetCDF via a git submodule
- p-refinement can now be selectively disabled and reenabled on a per-variable-group basis
- New error indicator for Empirical Interpolation Method (EIM) reduced-basis approximations
- `Elem::quality(ASPECT_RATIO)` is now defined for all element types
- Added option to use absolute instead of relative tolerances in linear solvers
- Added the capability to "upgrade" a vector's ghosting and/or projection settings in subsequent `System::add_vector()` calls
- Added `MeshRefinement::allow_unrefined_patches()` option
- Refactored iterator declarations and added more range methods in `MeshBase`: every filtered iterator that could be previously accessed with a `...begin()` and `...end()` method now also has a `...range()` method suitable for range-based for loops
- Better PETSc versioning macros for use with pre-release PETSc commits
- Build system now sets `CFLAGS_OPT`, etc. environment variables, not just `CXXFLAGS` and `CPPFLAGS` versions
- Many more unit tests for ExodusII I/O and for element refinement APIs
- Assorted fixes
  - Fixed Exodus I/O of side sets on 2D triangles embedded in 3D space
  - Fixed Nemesis dimension variable output
  - Fixed Nedelec one `FE` compatibility with `Tri7` and `Tet14` geometric elements
  - Fixed multiple issues with mesh refinement on meshes with certain types of infinite elements
  - Fixed no-parameters case and improved print statements in EIM code
  - Fixed typos in code comments
  - Minor fixes in vector FE example programs
- Assorted optimizations
  - An unnecessary pointer-indirection layer has been removed from `DiffContext`
  - Local matrix allocation in `DiffContext` objects is now optional
  - Minor optimizations have been made to Hilbert-curve global indexing calculations

Refs https://github.com/idaholab/moose/pull/25290 - we'll want to reenable those tests with new golds after this is safely in.

This fixes https://github.com/idaholab/moose/issues/24631

This fixes https://github.com/idaholab/moose/issues/25007 in my tests

Commit:
Simply testing for Conda MOOSE

Remove most testing being done during Conda MOOSE package creation.
Testing will be implemented as a separate target. This will allow
for quicker package creation when things go awry.

Closes #25369

cticenhour reviewed (dismissed):
Pending separate target being available.

milljm commented:
To be used with: https://github.com/idaholab/civet_recipes/pull/1448

Commit:
Remove build strings, bump SDK

Since this PR is a simple one, we might as well perform other
maintenance we've been wanting to do for a while.

Bump SDK build version to 10.15.

Remove build strings from packages that we only ever bump a date and
not a build number.

Commit:
Wasp needs build strings

Basically, dependency packages listed in versioner.yaml require
strings.

TODO: Even if the above is true, find a way to not require build
      strings. When we start to add tools, test-tools to this list
      (something I think we are going to need for another project
      I am working on), this will need to be solved.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1742884/) on c888f11 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1742884/) on c888f11 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1742884/) on c888f11 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
hard code 'build_0'

If we don't, we get hashes.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1743317/) on eb9f898 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1743317/) on eb9f898 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1743317/) on eb9f898 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Apptainer build rocky9](https://civet.inl.gov/job/1743320/) on eb9f898 : invalidated by @cticenhour

CIVET claimed python 3.10 and clang-tools were not available.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1743342/) on eb9f898 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25370/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1743360/) on eb9f898 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>507928</th>
      <th colspan=4>#25370 eb9f89</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.66%</td>
      <td>85.65%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>90468</td>
      <td>90465</td>
      <td>-3</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15151</td>
      <td>15154</td>
      <td>+3</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25370/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25370/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25370/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1746197/) on 841a3a0 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1746197/) on 841a3a0 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1746197/) on 841a3a0 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour reviewed (dismissed):
Looks good to me - tentative approval. Given last week, I would like a second set of eyes as well - @loganharbour?

loganharbour reviewed (dismissed):
Looks good. 

Let's coordinate on when to merge this so that we're all on board this time

roystgnr committed:
Update libMesh submodule

- NetCDF upgrade to 4.9.2, now incorporating NetCDF via a git
  submodule
- p-refinement can now be selectively disabled and reenabled on a
  per-variable-group basis
- New error indicator for Empirical Interpolation Method (EIM)
  reduced-basis approximations
- `Elem::quality(ASPECT_RATIO)` is now defined for all element types
- Added option to use absolute instead of relative tolerances in
  linear solvers
- Added the capability to "upgrade" a vector's ghosting and/or
  projection settings in subsequent `System::add_vector()` calls
- Added `MeshRefinement::allow_unrefined_patches()` option
- Refactored iterator declarations and added more range methods in
  `MeshBase`: every filtered iterator that could be previously
  accessed with a `...begin()` and `...end()` method now also has a
  `...range()` method suitable for range-based for loops
- Better PETSc versioning macros for use with pre-release PETSc
  commits
- Build system now sets `CFLAGS_OPT`, etc. environment variables,
  not just `CXXFLAGS` and `CPPFLAGS` versions
- Many more unit tests for ExodusII I/O and for element refinement
  APIs
- Assorted fixes
  - Fixed Exodus I/O of side sets on 2D triangles embedded in 3D
    space
  - Fixed Nemesis dimension variable output
  - Fixed Nedelec one `FE` compatibility with `Tri7` and `Tet14`
    geometric elements
  - Fixed multiple issues with mesh refinement on meshes with
    certain types of infinite elements
  - Fixed no-parameters case and improved print statements in EIM
    code
  - Fixed typos in code comments
  - Minor fixes in vector FE example programs
- Assorted optimizations
  - An unnecessary pointer-indirection layer has been removed from
    `DiffContext`
  - Local matrix allocation in `DiffContext` objects is now optional
  - Minor optimizations have been made to Hilbert-curve global
    indexing calculations

Refs #0

Refs #25290 - we'll want to reenable those tests with new golds after
this is safely in.

This fixes #24631

This fixes #25007 in my tests

roystgnr committed:
Update conda + docker configs for new libMesh

roystgnr committed:
Update versioner hashes

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit fd900b54a02b9d1b99f0a4371d4acb4761e57b0f
+Subproject commit ceff84386bdc83d87e9eb98ac3a9b745711e415a
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1753735/) on 4df8a4d wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1753735/) on 4df8a4d wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1753735/) on 4df8a4d wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour commented on conda/wasp/meta.yaml:
```diff
@@ -6,7 +6,7 @@
 # ALSO: Follow the directions in scripts/tests/versioner_hashes.yaml
 {% set build = 0 %}
 {% set strbuild = "build_" + build|string %}
-{% set version = "2023.08.17" %}
```
I don't believe that we needed to update wasp, but that's fine

cticenhour replied:
👍🏻

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/1754195/) on 4df8a4d wanted to post the following:

View timings [here](https://mooseframework.inl.gov/docs/PRs/25370/timing)

This comment will be updated on new commits.

roystgnr commented:
Is that valgrind phase field test just hitting a timeout repeatedly?

loganharbour commented:
> Is that valgrind phase field test just hitting a timeout repeatedly?

Yeah. https://github.com/idaholab/moose/pull/25420 will resolve it

cticenhour commented:
LGTM @loganharbour - I will leave the final merge to you.
